tests/integration/targets/s3_metrics_configuration/tasks/main.yml

```
@@ -0,0 +1,242 @@
+      security_token: '{{ security_token | default(omit) }}'
+      region: '{{ aws_region | default(omit) }}'
+    s3_metrics_configuration:
+      aws_access_key: '{{ aws_access_key | default(omit) }}'
+      aws_secret_key: '{{ aws_secret_key | default(omit) }}'
+      security_token: '{{ security_token | default(omit) }}'
+      region: '{{ aws_region | default(omit) }}'
+  collections:
+    - amazon.aws
+  block:
+
+  # ============================================================
+  - name: Try to create metrics configuration for non-existing bucket
+    s3_metrics_configuration:
+      bucket_name: '{{ test_bucket }}'
+      id: 'EntireBucket'
+      state: present
+    register: result
+    ignore_errors: yes
+
+  - assert:
+      that:
+      - result is failed
+
+  # ============================================================
+  - name: Create simple s3_bucket to act upon it
+    s3_bucket:
+      name: '{{ test_bucket }}'
+      state: present
+    register: output
+
+  - assert:
+      that:
+      - output is changed
+      - output.name == test_bucket
+
+  # ============================================================
+  - name: Create a metrics configuration under check mode
+    s3_metrics_configuration:
+      bucket_name: '{{ test_bucket }}'
+      id: 'EntireBucket'
+      state: present
+    check_mode: yes
+    register: result
+
+  - include_tasks: './s3_metrics_info.yml'
+
+  - assert:
+      that:
+      - result is changed
+      - metrics_info | selectattr('Id', 'search', 'EntireBucket') | list | length == 0
+
+  # ============================================================
+  - name: Create a metrics configuration that enables metrics for an entire bucket
+    s3_metrics_configuration:
+      bucket_name: '{{ test_bucket }}'
+      id: 'EntireBucket'
+      state: present
+    register: result
+
+  - include_tasks: './s3_metrics_info.yml'
+
+  - assert:
+      that:
+      - result is changed
+      - metrics_info | selectattr('Id', 'search', 'EntireBucket') | list | length == 1
+
+  # ============================================================
+  - name: Create a metrics configuration idempotency under check mode
+    s3_metrics_configuration:
+      bucket_name: '{{ test_bucket }}'
+      id: 'EntireBucket'
+      state: present
+    check_mode: yes
+    register: result
+
+  - assert:
+      that:
+      - result is not changed
+
+  # ============================================================
+  - name: Create a metrics configuration idempotency
+    s3_metrics_configuration:
+      bucket_name: '{{ test_bucket }}'
+      id: 'EntireBucket'
+      state: present
+    register: result
+
+  - assert:
+      that:
+      - result is not changed
+
+  # ============================================================
+  - name: Put a metrics configuration that enables metrics for objects starting with a prefix
+    s3_metrics_configuration:
+      bucket_name: "{{ bucket_name }}"
```

```suggestion
      bucket_name: "{{ test_bucket }}"
```
And several additional usages below

---

good spot